target/linux/ipq806x/files-4.14/arch/arm/boot/dts/qcom-ipq8064-qxwlan-e5200.dts

```diff
@@ -0,0 +1,290 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+ *
+ * Copyright (c) 2019 Peng Zhang <sd20@qxwlan.com>
+ *
+*/
+
+#include "qcom-ipq8064-v1.0.dtsi"
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "Qxwlan E5200";
+	compatible = "qxwlan,e5200", "qcom,ipq8064";
+
+	memory@0 {
+		reg = <0x42000000 0x1e000000>;
+		device_type = "memory";
+	};
+
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+		rsvd@41200000 {
+			reg = <0x41200000 0x300000>;
+			no-map;
+		};
+	};
+
+	aliases {
+		serial0 = &gsbi4_serial;
+		mdio-gpio0 = &mdio0;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	soc {
+		pinmux@800000 {
+			button_pins: button_pins {
+				mux {
+					pins = "gpio54", "gpio65";
+					function = "gpio";
+					drive-strength = <2>;
+					bias-pull-up;
+				};
+			};
+
+			led_pins: led_pins {
+				mux {
+					pins = "gpio7", "gpio8", "gpio9","gpio26", "gpio53";
+					function = "gpio";
+					drive-strength = <2>;
+					bias-pull-up;
+				};
+			};
+
+			i2c4_pins: i2c4_pinmux {
+				pins = "gpio12", "gpio13";
+				function = "gsbi4";
+				bias-disable;
+			};
+
+			spi_pins: spi_pins {
+				mux {
+					pins = "gpio18", "gpio19", "gpio21";
+					function = "gsbi5";
+					drive-strength = <10>;
+					bias-none;
+				};
+			};
+			nand_pins: nand_pins {
+				mux {
+					pins = "gpio34", "gpio35", "gpio36",
+					       "gpio37", "gpio38", "gpio39",
+					       "gpio40", "gpio41", "gpio42",
+					       "gpio43", "gpio44", "gpio45",
+					       "gpio46", "gpio47";
+					function = "nand";
+					drive-strength = <10>;
+					bias-disable;
+				};
+				pullups {
+					pins = "gpio39";
+					bias-pull-up;
```

Comment:
It's a bit odd that you set it to bias-disable in the "mux" node above. You should be able to remove the gpio39 from the list in "mux" above and add the function = "nand" + drive-strength to this node.

Reply:
I saw the qsdk and openwrt nodes seem to be described as such, so I also follow the same description.
If this is not correct, then how can I modify this node?
please help me,Thanks!

Reply:
Well, in that case what does your contact at Qualcomm say about it? Either they must have a reason or they don't.

Reply:
Sorry, I can't contact Qualcomm, I don't have their contact information.

Reply:
@Ansuel @pkgadd maybe you've an idea how to handle this?

Reply:
@ynezz I actually found out what @chunkeey  meant.
As he said it's strange to put bias-disable in the mux.... The best thing to do would be create a separate group with bias-disable. I'm converting every dts we have to follow this rule as this is what the kernel guy wants.

Reply:
Here an example
https://github.com/Ansuel/openwrt/commit/a608ffe6e487d0466f245893cae4d9c0cde60d94

Reply:
Here what @chunkeey suggested to do... Read some documentation and having a mux with no bias definition looks like it gets ignored by the driver
https://github.com/Ansuel/openwrt/commit/12b0bf3a5d9b72726f662d8573636afeade52e91 
This is how it should be defined... Tested on my r7800 and works like usual... (if it wasn't... then i wouldn't be able to boot AHAHAH)